clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -234,7 +237,8 @@ public long maximumTimeToWait(long currentTimeMs) {
      * When consumer polls, we need to reset the pollTimer.  If the poll timer has expired, we rejoin only when the
      * member is in the {@link MemberState#UNSUBSCRIBED} state.
      */
-    public void resetPollTimer() {
+    public void resetPollTimer(final long pollMs) {
+        pollTimer.update(pollMs);
```

Comment:
note: pollMs is the time user invoke consumer#poll.  This is better than taking time.millisecond() because there could be some discrepancy.